web/styles/compose.css

```diff
@@ -535,11 +535,49 @@ textarea.new_message_textarea,
     }
 }
 
-input.recipient_box {
-    margin: 0;
-    padding: 0 6px;
-    height: auto;
+#compose_recipient_box {
+    /* Recipient box is the div container that has input and button */
     border-radius: 3px;
+    display: flex;
+    flex-grow: 1;
+    background: hsl(0deg 0% 100%);
+
+    &.focused {
+        outline: 0;
+        border: 1px solid hsl(0deg 0% 67%);
+        box-shadow: none;
+    }
+
+    #stream_message_recipient_topic,
+    #recipient_box_new_topic_button {
+        background: none;
+        border: none;
+        color: inherit;
+    }
+
+    /* Styles for input in the recipient_box */
+    #stream_message_recipient_topic {
+        flex-grow: 1;
+        box-shadow: none;
+    }
+
+    /* Styles for new topic button in the recipient_box */
+    #recipient_box_new_topic_button {
+        flex-shrink: 0;
+        /* The button looks better if it have smaller radius than the parent */
+        border-radius: 2px;
```

Comment:
These are small values, but did you eyeball that value? It's pretty easy to calculate harmonious border-radius values for nested elements: https://www.30secondsofcode.org/css/s/nested-border-radius/

Reply:
Thanks for sharing that, it's really interesting. Here's the quote from the post you provided:

> This can be expressed mathematically as innerRadius + distance = outerRadius, or more simply R1 + D = R2.

 I did exactly the same thing, just without the theoretical background. In our case, we have 3 pixels on the parent, 1 pixel for the margin, and the resulting value for the inner radius is 2 pixels.